docs/source/usage/torch_compile.rst

```diff
@@ -0,0 +1,200 @@
+    Throughput: 44.39 requests/s, 22728.17 total tokens/s, 11364.08 output tokens/s
+
+    $ # 2. Run the same setting with profiling
+    $ VLLM_LOG_BATCHSIZE_INTERVAL=1.0 python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Meta-Llama-3-8B --load-format dummy --num-scheduler-steps 64
+    INFO 12-10 15:42:47 forward_context.py:58] Batchsize distribution (batchsize, count): [(256, 769), (232, 215), ...]
+
+    $ # 3. The most common batch sizes are 256 and 232, so we can compile the model for these two batch sizes
+    $ python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Meta-Llama-3-8B --load-format dummy --num-scheduler-steps 64 -O "{'level': 3, 'candidate_compile_sizes': [232, 256]}"
+    init engine (profile, create kv cache, warmup model) took 87.18 seconds
+    Throughput: 46.11 requests/s, 23606.51 total tokens/s, 11803.26 output tokens/s
+
+Note that ``torch.compile`` only helps to accelerate the model forwarding. To see the benefit, please make sure GPUs are already busy executing the model; otherwise, the benefit will be hidden because GPUs are idle. That's why we have added ``--num-scheduler-steps 64`` to the command line arguments when benchmarking the throughput.
+
+How does it work?
+-----------------
+
+For text-only models, we compile the part of the model from input token IDs to final hidden states, excluding the LM head and logits processing.
+
+For multi-modality models, we compile the text-only part of the model from input embeddings to final hidden states, excluding the vision encoder part and the part of merging multi-modality embeddings with text embeddings.
+
+By carefully compiling the main computation of the model, we can avoid unnecessary compilation time and achieve better performance.
+
+Supported Models
+----------------
+
+Most models in vLLM are supported by ``torch.compile``. You should see logs like ``torch.compile takes 19.37 s in total`` in the server logs when you enable ``torch.compile``. If a model does not support ``torch.compile`` but you enable it, there will be a warning ``torch.compile is turned on, but the model does not support it``, and the ``torch.compile`` configurations will be ignored. If you want to get this model supported, please file an issue.
+
+The following models are currently not supported by ``torch.compile``, because their computation graphs are too dynamic to compile:
+
+- ``InternLM2VEForCausalLM``, ``InternVLChatModel``
+- cross-attention models like ``MllamaForConditionalGeneration`` and ``BartForConditionalGeneration``
+
+The following models should be supported by ``torch.compile`` in the future, but not supported yet due to bandwidth limitations:
+
+- ``Mamba`` related models
+- ``ChameleonModel``, ``ChatGLMModel``, ``DbrxModel``, ``DeepseekModel``, ``MixtralModel``, ``Olmo2Model``, ``Phi3SmallModel``, ``StableLMEpochModel``
```

Comment:
@DarkLight1337 I checked all the models, and these should be the list of unsupported models. I will also update the `add models` page to show how to add support for new models shortly.